runtime/gc_base/ReferenceChainWalker.cpp

```diff
@@ -646,7 +646,7 @@ MM_ReferenceChainWalker::doStackSlot(J9Object **slotPtr, void *walkState, const
 	/* Only report heap objects */
 	if (isHeapObject(slotValue) && !_heap->objectIsInGap(slotValue)) {
 		if (J9_STACKWALK_SLOT_TYPE_JNI_LOCAL == ((J9StackWalkState *)walkState)->slotType) {
-			doSlot(slotPtr, J9GC_ROOT_TYPE_JNI_LOCAL, -1, (J9Object *)walkState);
+			doSlot(slotPtr, J9GC_ROOT_TYPE_JNI_LOCAL, -1, NULL);
 		} else {
 			doSlot(slotPtr, J9GC_ROOT_TYPE_STACK_SLOT, -1, (J9Object *)walkState);
```

Comment:
I thought the cast was suspicious when I saw #18379, `NULL` seems more reasonable, but why should the call on line 651 not also pass `NULL`?

Reply:
A valid question. In theory, callbacks being aware that the type is STACK_SLOT could safely cast the referrer parameter back to walkState type and do something about that info/structure (print stack slot address etc), but I don't know if they do.

I'm ok with passing NULL, but not don't have the whole picture (history and awareness of current callbacks) to say it's really safe.

Reply:
If we pass NULL for STACK_SLOT, it will break the expectation in `jvmtiHeap10.c::processStackRoot()` function.